UWPPackageInstaller/MainPage.xaml.cs

```diff
@@ -1,385 +1,160 @@
-
-            //Normal Code:
-            //If the device is on the creators update or later, install progress is shown in the action center and App UI
-            //Otherwise, all progress is shown in the App's UI.
-            if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 4))
-            {
-                showProgressInNotification();
-            }
-            else
-            {
-                showProgressInApp();
-            }
-
-
-
+            InstallButton.Visibility = Visibility.Collapsed;
+            CancelButton.Visibility = Visibility.Collapsed;
 
+            var (url, filename) = _appDownloadInfo;
+            downloadAndInstall(url, filename);
         }
 
-
-        private async void showProgressInApp()
+        private async void downloadAndInstall(Uri fileToDownload, string packageName)
         {
-            installProgressBar.Visibility = Visibility.Visible;
-            installValueTextBlock.Visibility = Visibility.Visible;
-            PackageManager pkgManager = new PackageManager();
-            Progress<DeploymentProgress> progressCallback = new Progress<DeploymentProgress>(installProgress);
-            string resultText = "Nothing";
+            InstallProgressBar.Visibility = Visibility.Visible;
+            InstallValueTextBlock.Visibility = Visibility.Visible;
 
-            notification.showInstallationHasStarted(packageInContext.Name);
-            if (dependencies != null && dependencies.Count > 0)
-            {
-                try
-                {
-                    var result = await pkgManager.AddPackageAsync(new Uri(packageInContext.Path), dependencies, DeploymentOptions.ForceTargetApplicationShutdown).AsTask(progressCallback);
-                    checkIfPackageRegistered(result, resultText);
+            IProgress<DeploymentProgress> progressCallback = new Progress<DeploymentProgress>(installProgress);
+            string resultText = "Nothing";
 
-                }
-                catch (Exception e)
-                {
-                    resultText = e.Message;
-                }
+            Notification.ShowInstallationHasStarted(packageName);
+            var pkgRegistered = true;
 
-            }
-            else
+            try
             {
-                try
-                {
-
-                    var result = await pkgManager.AddPackageAsync(new Uri(packageInContext.Path), null, DeploymentOptions.ForceTargetApplicationShutdown).AsTask(progressCallback);
-                    checkIfPackageRegistered(result, resultText);
-                }
+                var addPackageOperation = _packageManager.AddPackageAsync(fileToDownload, null,
+                    DeploymentOptions.ForceApplicationShutdown | DeploymentOptions.ForceUpdateFromAnyVersion);
+                // There is no progress callback while downloading (that I have found)
+                PermissionTextBlock.Text = "Downloading files... Download progress is unknown.";
```

Comment:
This message feels like an error message. Maybe add something like "... Install will soon start automatically" or something